remotes/fixup.go

```diff
@@ -136,3 +179,55 @@ func checkBaseImage(baseImage *bundle.BaseImage) error {
 
 	return nil
 }
+
+// FixupEvent is an event that is raised by the Fixup Logic
+type FixupEvent struct {
+	SourceImage    string
+	DestinationRef reference.Named
+	Descriptor     *ocischemav1.Descriptor
+	EventType      FixupEventType
+	Message        string
+	Error          error
+	DigestTree     *DigestTree
+}
+
+// FixupEventType is the the type of event raised by the Fixup logic
+type FixupEventType string
+
+const (
+	// FixupEventTypeCopyImageStart is raised when the Fixup logic starts copying an
+	// image
+	FixupEventTypeCopyImageStart = FixupEventType("CopyImageStart")
+
+	// FixupEventTypeCopyImageEnd is raised when the Fixup logic stops copying an
+	// image. Error might be populated
+	FixupEventTypeCopyImageEnd = FixupEventType("CopyImageEnd")
```

Comment:
Can we (do we want to) correlate `Start` vs `End` for all these? Are they serialised or might they interleve?

Reply:
They are correlated by the other fields (CopyImage is correlated by SourceImage, CopyDescriptor/MounDescriptor are correlated by Descriptor). For now, everything is serialized, but I suspect we can introduce some level of parallellism, as soon as start/end are correlated.

Reply:
Events have changed (a lot) since your review. marking as resolved